lib/foreman/renderer/scope/macros/inputs.rb

```diff
@@ -25,6 +25,39 @@ def input(name)
               raise UndefinedInput.new(s: name)
             end
           end
+
+          apipie :method, 'Find and returns resource object from the template input value' do
+            required :name, String, desc: 'name of the template input'
+            raises error: UndefinedInput, desc: 'when there is no input with such name defined for the current template'
+            raises error: WrongInputValueType, desc: 'when the input value type is not "resource"'
+            returns Object, desc: 'The resource object'
+            example 'input_resource("hostgroup") #=> "ActiveRecord object"'
+          end
+          def input_resource(name)
+            @input = template.template_inputs&.find_by_name(name)
+
+            raise UndefinedInput.new(s: name) unless @input
+            raise WrongInputValueType.new(name: name, type: @input.value_type) if @input.value_type != 'resource'
+
+            return @input.preview(self) if preview?
+
+            find_resource
+          end
+
+          private
+
+          def resource_klass
+            @input.resource_type.constantize
```

Comment:
Why do we not rescue the wrong resource class? It might happen in some cases (removed plugin).
We should probably rescue from NameError here.

Reply:
What should raise instead of `NameError`?

Reply:
Some rendering error that would be more explanatory for the user ? :thinking:
Note I didn't try and didn't read the message that will get to the user, if you think it's user friendly enough (I'd even be ok with untranslated, if it looks ok), just say so and let's merge it :)

Reply:
Yeah you are right, i was thinking that `NameError` is good enough, but better explanation for user is always better. Updated, now raising `Foreman::Renderer::Errors::UnknownResource` error with message `"Unkown 'Klass' resource class"`